pkg/repository/prisma/dbsqlc/workflow_runs.sql

```diff
@@ -1415,3 +1415,22 @@ WHERE
     sre."workflowRunId" = @workflowRunId::uuid
 ORDER BY
     sre."id" DESC;
+
+-- name: GetFailureDetails :one
+SELECT
+	wr."status",
+	wr."id",
+	jr."status" as "jrStatus",
+	sr."status" as "srStatus",
+	sr."cancelledReason",
+	sr."error"
+FROM "WorkflowRun" wr
+JOIN
+	"JobRun" jr on jr."workflowRunId" = wr."id"
+JOIN
+	"StepRun" sr on sr."jobRunId" = jr."id"
+WHERE
+	wr."status" = 'FAILED' AND
+	wr."id" = @workflowRunId::uuid AND
+    wr."tenantId" = @tenantId::uuid
+LIMIT 1;
```

Comment:
This is a LIMIT 1 but you are joining job run and step run, of which there could be many, so you might be selecting a succeeded step run in a workflow with many steps.